### ISSUE TITLE
Some draft idea about the project

### ISSUE THREAD

**Issue by chengkaiyang2025** [chengkaiyang2025/Summer-2025-ECE-597-Group11#1]
Build very basic classifier using traditional ML techniques Codebase and documentation 
for reproducibility and future extension of the project.

# Label the dataset
- Building the training dataset and label the dataset.
- Or find the open-source dataset for different kind of spam.
- Using Clustering to find similar spam, and add label to them.

# Features
- Email length
- Number of URLs
- Proportion of suspicious words (verify, prize, urgent, discount, account, bank)
- Sentiment  "danger", "hurry", "guarantee", "risk"
- Use of punctuation marks

# add more Feature like
- TF-IDF
- 3-gram to find suspicious words or term(unusual activity,urgently required)

# Topic Modeling
# Online checker
https://www.warmupinbox.com/email-spam-words-checker/
https://www.salesforge.ai/tools/spam-checker

# Method
- Decision tree
- Linear regression + softmax
- NN
